CMakeLists.txt

```diff
@@ -67,7 +67,7 @@ option(WITH_DLT_TESTS         "Set to ON to build src/test binaries"
 option(WITH_DLT_UNIT_TESTS    "Set to ON to build gtest framework and tests/binaries"                            OFF)
 option(WITH_DLT_QNX_SYSTEM    "Set to ON to build QNX system binary dlt-qnx-system"                              OFF)
 
-set(DLT_IPC "FIFO" CACHE STRING "UNIX_SOCKET,FIFO")
+set(DLT_IPC "FIFO" CACHE STRING "UNIX_SOCKET,FIFO,QNX_MESSAGE")
```

Comment:
If the user set to QNX_MESSAGE, we would verify first that it is compiled under QNX, otherwise raise an error and return already. You could do something similar to line 161. Then you don't need to consider QNX and non-QNX inside headers and codes.